### PR TITLE
Connect 2005 api v1 status

### PR DESCRIPTION
**Pull request by Majkluss**

### CODE REVIEW

File: prusa/link/web/main.py [hunk 1]
```diff
@@ -51,6 +51,15 @@
     run directly from your printer's internal storage)."""
 
 
+def filter_none(dictionary):
```
Comment (ondratu) [inline prusa/link/web/main.py:54]: There is some similar function in SDK, please check it.

File: prusa/link/web/main.py [hunk 2]
```diff
@@ -134,6 +143,58 @@ def api_info(req):
...
+    status = {}
+
+    status_printer = {
+        "temp_nozzle": tel.temp_nozzle,
+        "temp_bed": tel.temp_bed,
+        "axis_z": tel.axis_z
+    }
+
+    # X and Y axes data are available only when not printing
+    if job.job_state is JobState.IDLE:
```
Comment (ondratu) [inline prusa/link/web/main.py:165]: no, if job.job_state is not PRINTING or BUSY. The axis are relevant when printer is not moving.

File: prusa/link/web/main.py [hunk 3]
```diff
@@ -134,6 +143,58 @@ def api_info(req):
...
+        status_printer["axis_x"] = tel.axis_x
+        status_printer["axis_y"] = tel.axis_y
+    else:
+        progress = (tel.progress or 0) / 100.0 if is_printing else None
```
Comment (ondratu) [inline prusa/link/web/main.py:169]: when job state is not IDLE it is always printing?

File: prusa/link/web/main.py [hunk 4]
```diff
@@ -134,6 +143,58 @@ def api_info(req):
...
+        status["job"] = filter_none(status_job)
+
+    status_printer = filter_none(status_printer)
+    if status_printer:
+        status["printer"] = status_printer
+
+    if transfer.in_progress:
+        status_transfer = {
+            "id": transfer.transfer_id,
+            "time_elapsed": int(transfer.time_elapsed())
```
Comment (ondratu) [inline prusa/link/web/main.py:188]: there is probably same problem as in SDK

**Review by ondratu — CHANGES_REQUESTED** [review]

File: prusa/link/web/main.py (hunk 1 above)
Comment (Majkluss) [reply prusa/link/web/main.py:54]: Used filter_null from SDK

**Review by ondratu — CHANGES_REQUESTED** [review]

**Review by ondratu — APPROVED** [review]

**Pull request merged**